open-craft/ansible-playbooks
# FAL-718 fix elasticsearch deployment

[FAL-718](https://tasks.opencraft.com/browse/FAL-718)

- fix backward compatibility with existing elasticsearch setups (conditionally enable the new password/keystore/kibana things)
- support elasticsearch 7 (via extra cfg; es 7 requires `discovery.type` set)
- add standard .tool-versions file with a supported python version (so i stop accidentally using python 3.9 which isn't supported here yet)

**Test instructions**:

This was used to deploy the elasticsearch-labxchange-ci instance (see https://github.com/open-craft/ansible-secrets/pull/246 ), so just need to verify that PR.

Also read through and sanity check all the changes - it needs to work with the new setup from https://github.com/open-craft/ansible-playbooks/pull/188 , as well as the original setup for elasticsearch 1.5 and dedicated es/memcached instances for ocim instances.  Ideally, we would run the playbooks against one of the new ES servers with the kibana setup, and one of the older ES servers used with ocim instances, but not sure if that's a good idea - if there's a bug, it will break the production servers. :grimacing: 

**Reviewers**:
